### PR TITLE
V1.45

### PR DESCRIPTION
**Pull request by zhaozg**
close https://github.com/luvit/luv/issues/646

### CODE REVIEW

**Comment by squeek502** [comment]
My suggestion would be to remove the `update lua to v5.4.6` and `update lua-compat-5.3 to HEAD` commits from this branch to see if updating Lua is the cause. If it is, then we should just do that separately from updating Libuv.

**Comment by zhaozg** [comment]
Ok, change finished.

**Comment by squeek502** [comment]
Created a pull request with the remaining stuff: https://github.com/zhaozg/luv/pull/5

**Review by squeek502 — APPROVED** [review]

**Comment by zhaozg** [comment]
Good jobs, Let wait appveyor report status.

**Comment by zhaozg** [comment]
failed https://ci.appveyor.com/project/racker-buildbot/luv/builds/47095901

**Comment by squeek502** [comment]
Strange, will figure that out in a bit. I'm currently on Linux reporting the valgrind libuv stuff :smile:

**Comment by zhaozg** [comment]
> Strange, will figure that out in a bit. I'm currently on Linux reporting the valgrind libuv stuff 😄

Maybe lua cast it as signed number. I do some check.

**Comment by squeek502** [comment]
Looks like you're right. Changing `luv_get_available_memory` to use `lua_pushnumber` should fix it I think.

**Comment by squeek502** [comment]
Looks like `pushunsigned` won't work, since the `lua-compat-5.3` implementation just casts to `lua_Integer` anyway:

https://github.com/lunarmodules/lua-compat-5.3/blob/8f8e4c6adb43e107f5902e784ef207dc3c8ca06b/c-api/compat-5.3.h#L258-L259

I think `lua_pushnumber` is okay, it's what `luv_get_free_memory`, `luv_get_total_memory`, etc use.

**Comment by zhaozg** [comment]
ok, I will back in 20 minutes, outsides now.

**Review by squeek502 — CHANGES_REQUESTED** [review]
Found a bad bug in the code I wrote, will submit a fix in a few minutes.

**Comment by squeek502** [comment]
https://github.com/zhaozg/luv/pull/6

**Review by squeek502 — APPROVED** [review]

File: rockspecs/luv-scm-0.rockspec [hunk 1]
```diff
@@ -79,6 +79,9 @@ build = {
             'userenv';
             'ws2_32';
             'advapi32';
+            'Dbghelp';
+            "Ole32";
+            "Shell32";
```
Comment (squeek502) [inline rockspecs/luv-scm-0.rockspec:84]: Why are these needed?

Comment (zhaozg) [reply rockspecs/luv-scm-0.rockspec:84]: to fix https://ci.appveyor.com/project/racker-buildbot/luv/builds/47096188/job/9j5t4vt905amaa2f#L1222
